lib/puppet/provider/ibm_pkg/imcl.rb

```diff
@@ -69,17 +69,46 @@ def imcl_command_path
     @imcl_command_path
   end
 
+
+  # searches for installed.xml in potential appDataLocation dirs
+  #
+  # @return [Array] array of potential paths to installed.xml
+  def self.installed_xml_paths
+    require 'find'
+    glob = Dir.glob('/home/*/var/ibm/') # returns list of dirs
+
+    installed_xml_paths = Array.new
+
+    begin
+      # pass in the glob and the default appDataLocation (/var/ibm/)
+      Find.find("/var/ibm/", *glob) do |path|
+        installed_xml_paths.push(path) if path.match(/InstallationManager\/installed.xml/)
+      end
+    rescue Errno::ENOENT => e
+      raise("There was a problem finding installed.xml: #{e}")
+    end
+
+    installed_xml_paths
+  end
+
   # returns a file handle by opening the install file
   # easier to mock when extracted to method like this
   #
   # @return [String] path to installed.xml file
-  def installed_file
-    if resource[:user] == 'root'
-      xml_path = '/var/ibm/InstallationManager/installed.xml'
+  def self.installed_file(user)
+    if installed_xml_paths.length > 1
+      installed_xml_paths.each do |path|
+        if user == 'root'
+          return path if path.match(/^\/var\//)
+        else
+          return path if path.match(/\/#{user}\//)
```

Comment:
If you only care about the `installed.xml` file for the cases that match `user`, then why go through the exercise of getting all of the possible `installed.xml` paths? Why not look for `/var/ibm/` if `user == root` and `/home/#[user}/var/ibm` if `user` is anything else?